marshal.go

```
@@ -2404,12 +2660,139 @@ type UDTUnmarshaler interface {
 	UnmarshalUDT(name string, info TypeInfo, data []byte) error
 }
 
-func marshalUDT(info TypeInfo, value interface{}) ([]byte, error) {
-	udt := info.(UDTTypeInfo)
+type udtCQLType struct {
+	types *RegisteredTypes
+}
```

I'm still not sure what the goal of the UDT registration is with this API, I don't think users can leverage this API to replace the usage of `UDTMarshaler` and `UDTUnmarshaler` right? I wonder if we should just return an error when a user tries to register a UDT at the moment and create a new ticket to add proper UDT support to this API later in a 2.x minor release.

---

Also, proper UDT support will have to include reading the system tables to get the information about the fields of the UDT so a UDT marshaler will need access to a connection or some kind of API that allows the marshaler to obtain that information from the session.

With that in mind it is very likely that the interface for a UDT marshaler will be different from the other ones or alternatively we only allow users to provide "go type <-> cql user type" mappings instead of a complete udt marshaler. This is another reason to just return an error if users try to do anything with UDTs when using this new API for now.

---

Yeah I envision a `RegisterUDT` in the future but we can wait until post-2.0 for that.